src/db.rs

```diff
@@ -1753,46 +1788,39 @@ impl DBWithThreadMode<MultiThreaded> {
     /// Creates column family with given name and options
     pub fn create_cf<N: AsRef<str>>(&self, name: N, opts: &Options) -> Result<(), Error> {
         let inner = self.create_inner_cf_handle(name.as_ref(), opts)?;
-        self.cfs
-            .cfs
-            .write()
-            .unwrap()
-            .insert(name.as_ref().to_string(), ColumnFamily { inner });
+        self.cfs.cfs.write().unwrap().insert(
+            name.as_ref().to_string(),
+            Arc::new(UnboundColumnFamily { inner }),
+        );
         Ok(())
     }
 
     /// Drops the column family with the given name by internally locking the inner column
     /// family map. This avoids needing `&mut self` reference
     pub fn drop_cf(&self, name: &str) -> Result<(), Error> {
-        let inner = self.inner;
         if let Some(cf) = self.cfs.cfs.write().unwrap().remove(name) {
-            unsafe {
-                ffi_try!(ffi::rocksdb_drop_column_family(inner, cf.inner));
-            }
-            Ok(())
+            self.drop_column_family(cf.inner, cf)
         } else {
             Err(Error::new(format!("Invalid column family: {}", name)))
         }
     }
 
     /// Returns the underlying column family handle
-    pub fn cf_handle(&self, name: &str) -> Option<BoundColumnFamily> {
+    pub fn cf_handle(&self, name: &str) -> Option<Arc<BoundColumnFamily>> {
         self.cfs
             .cfs
             .read()
             .unwrap()
```

Comment:
```suggestion
            .ok()?
```

Reply:
@aleksuss hi! thanks for continued review while you're very busy... :)

in short, I think RwLock's {`.read()`/`.write()`} should be paired with `.unwrap()`.  This topic is called _lock poisoning_:

https://doc.rust-lang.org/nomicon/poisoning.html:

> However if the Mutex contained, say, a BinaryHeap that does not actually have the heap property, it's unlikely that any code that uses it will do what the author intended. As such, the program should not proceed normally. Still, if you're double-plus-sure that you can do something with the value, the Mutex exposes a method to get the lock anyway. It is safe, after all. Just maybe nonsense.

(Replace `BinaryHeap` here with our case: `BTreeMap`.)


Also, some relevant info:

- https://blog.rust-lang.org/2020/12/11/lock-poisoning-survey.html
- https://hackmd.io/@rust-libs/BJZxn8a3D#Which-of-these-is-closest-to-how-you-acquire-locks-Select-all-that-apply

Also, I've quickly skimmed wild usage (using our monorepo as sample..), too:

![image](https://user-images.githubusercontent.com/117807/126521257-ef5c5ea2-b68d-4a02-a459-012cd8d72209.png)

also, notice that rocksdb appears in the above screenshot, because I did `.unwrap()` at my previous pr like this: https://github.com/rust-rocksdb/rust-rocksdb/pull/506/files#r674132675